### DISCUSSION TITLE
Create a Publisher and Subscriber from the same participant

### DISCUSSION THREAD

**Discussion by brosito** [eProsima/Fast-DDS#4003] [General]
I am very new at FastDDS and I am struggling with some basic concepts. 
1) How create a publisher and subscriber from the same participant and topic? Every code sample shows the publisher and subscriber as different participants.
2) To avoid feedback loop, how can I make the subscriber ignore everything publish by its own participant?
Thank you!

**Comment by JLBuenoLopez** [comment]
@brosito 

1. You can create any number of Publishers/Subscribers and DataReaders/DataWriters within the same DomainParticipant connected with the same Topic if you want.
2. The [ignore local endpoints](https://fast-dds.docs.eprosima.com/en/latest/fastdds/property_policies/ignore_local_endpoints.html#ignore-local-endpoints) feature is what you are looking for.

**Reply by brosito** [reply]
Thank you @JLBuenoLopez-eProsima 
For questions 1, I understand that it can be done, but  when I try to do it by modifying the examples, I cant tell if it is working or not. I tried the ignore local end points but the issue persist. I think the way I am creating the subscriber under the same participant is wrong, but I cant find my mistake. Do you have a small example on how to do it? Thank you.

**Comment by brosito** [comment]
Hi @JLBuenoLopez-eProsima, I followed the examples provided in the repo. I added the ignore_local_point to the participant qos, then created the participant, then created the publisher and subscriber with that participant. then publisher created the writer, the subcriber created the reader. Ran the app, and every time I publish the subscriber gets the message. I am running both on the same terminal one program two threads. it is just not working. any suggestions.

**Reply by VWY-github** [reply]
Would like to see an example of this, and also an example for a "message manager" that deals with many topics with minimal code duplication.

**Reply by brosito** [reply]
Yes, please. Thank you.

**Comment by JesusPoderoso** [comment]
Hi @brosito, @VWY-github, thanks for using Fast DDS.

Unfortunately, after checking all the examples, you may have realized that all of them create publishers and subscribers in different participants.

I've quickly updated one of them in [this branch](https://github.com/eProsima/Fast-DDS/tree/testing_discussion_4003) to create them in a single participant.
The [second commit](https://github.com/eProsima/Fast-DDS/compare/master...testing_discussion_4003) includes the [ignore local endpoints](https://fast-dds.docs.eprosima.com/en/latest/fastdds/property_policies/ignore_local_endpoints.html#ignore-local-endpoints) feature, so you can check.

The following image represents the executed example before and after including (uncommenting) the ignore local endpoints feature, so you may observe that the subscriber does not receive any sample in the second case. Hope it helps!

![image](https://github.com/eProsima/Fast-DDS/assets/120394830/d4d02f56-1bcc-4b1d-879c-674ad8e8798d)

**Reply by brosito** [reply]
Hi @JesusPoderoso, thank you for providing an example with one participant. 

I tried to run your example but I get a DEPRECATED ERROR when compiling. Therefore, I copy your code into mine but there is not difference between commenting/uncommenting the ignoring local endpoints feature. 

I'll continue to troubleshoot.

**Reply by JesusPoderoso** [reply]
Hi @brosito, I've tested it and it builds on my side. I've rebased it on top of master and it still builds.
Please, make sure the proper branch is checked out, and the current Fast DDS dependencies have been set up (the latest v2.14.0 Fast DDS version was released a week ago, so try to update to [Fast DDS 2.14.x dependencies](https://fast-dds.docs.eprosima.com/en/latest/notes/versions.html))
